src/controller/python/chip/yaml/YamlParser.py

```diff
@@ -0,0 +1,351 @@
+#
+#    Copyright (c) 2022 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+from dataclasses import field
+import typing
+from chip import ChipDeviceCtrl
+from chip.tlv import float32
+import yaml
+import stringcase
+import chip.clusters as Clusters
+import chip.interaction_model
+import asyncio as asyncio
+import logging
+import math
+import chip.yaml.YamlUtils as YamlUtils
+
+
+logger = logging.getLogger('YamlParser')
+
+
+class ParsingError(Exception):
```

Comment:
Suggest descending from `ValueError` so you can catch the wider net without needing the full generic exception.